x-pack/plugins/fleet/server/routes/epm/index.ts

```diff
@@ -36,88 +37,110 @@ import {
 
 const MAX_FILE_SIZE_BYTES = 104857600; // 100MB
 
-export const registerRoutes = (routers: { rbac: FleetRouter; superuser: FleetRouter }) => {
-  routers.rbac.get(
+export const registerRoutes = (routers: {
+  fleetRouter: FleetAuthzRouter;
+  superuser: FleetRouter;
+}) => {
+  routers.fleetRouter.get(
     {
       path: EPM_API_ROUTES.CATEGORIES_PATTERN,
       validate: GetCategoriesRequestSchema,
-      options: { tags: [`access:${PLUGIN_ID}-read`] },
+      fleetAuthz: {
+        integrations: ['readPackageInfo'],
+      },
     },
     getCategoriesHandler
   );
 
-  routers.rbac.get(
+  routers.fleetRouter.get(
     {
       path: EPM_API_ROUTES.LIST_PATTERN,
       validate: GetPackagesRequestSchema,
-      options: { tags: [`access:${PLUGIN_ID}-read`] },
+      fleetAuthz: {
+        integrations: ['readPackageInfo'],
+      },
     },
     getListHandler
   );
 
-  routers.rbac.get(
+  routers.fleetRouter.get(
     {
       path: EPM_API_ROUTES.LIMITED_LIST_PATTERN,
       validate: false,
-      options: { tags: [`access:${PLUGIN_ID}-read`] },
+      fleetAuthz: {
+        integrations: ['readPackageInfo'],
+      },
     },
     getLimitedListHandler
   );
 
-  routers.rbac.get(
+  routers.fleetRouter.get(
     {
       path: EPM_API_ROUTES.STATS_PATTERN,
       validate: GetStatsRequestSchema,
-      options: { tags: [`access:${PLUGIN_ID}-read`] },
+      fleetAuthz: {
+        integrations: ['readPackageInfo'],
+      },
     },
     getStatsHandler
   );
 
-  routers.rbac.get(
+  routers.fleetRouter.get(
     {
       path: EPM_API_ROUTES.FILEPATH_PATTERN,
       validate: GetFileRequestSchema,
-      options: { tags: [`access:${PLUGIN_ID}-read`] },
+      fleetAuthz: {
+        integrations: ['readPackageInfo'],
+      },
     },
     getFileHandler
   );
 
-  routers.rbac.get(
+  routers.fleetRouter.get(
     {
       path: EPM_API_ROUTES.INFO_PATTERN,
       validate: GetInfoRequestSchema,
-      options: { tags: [`access:${PLUGIN_ID}-read`] },
+      fleetAuthz: {
+        integrations: ['readPackageInfo'],
+      },
     },
     getInfoHandler
   );
 
-  routers.superuser.put(
+  routers.fleetRouter.put(
     {
       path: EPM_API_ROUTES.INFO_PATTERN,
       validate: UpdatePackageRequestSchema,
-      options: { tags: [`access:${PLUGIN_ID}-all`] },
+      fleetAuthz: {
+        integrations: ['upgradePackages', 'writePackageSettings'],
```

Comment:
should the possible values be extracted to enum/constants? it is hard to see what are all the options

Reply:
the type here is `keyof fleetAuthz['integrations']` is not a plain string, autocomplete works in vscode and typescript will raise an error if you put a random string.